web-api/src/storage/postgres.rs

```diff
@@ -1263,3 +1272,60 @@ impl storage::Size for Storage {
         Ok(size)
     }
 }
+
+#[async_trait(?Send)]
+impl storage::IndexedProperties for Storage {
+    async fn load_schema(&self) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let schema = self.load_schema_with_transaction(&mut tx).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+
+    async fn extend_schema(
+        &self,
+        update: IndexedPropertiesSchemaUpdate,
+        max_properties: usize,
+    ) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let mut schema = self.load_schema_with_transaction(&mut tx).await?;
+        schema.update(&update, max_properties)?;
+        self.extend_postgres_schema(&mut tx, &update).await?;
+        //TODO self.elastic.extend_elasticsearch_mapping(&update).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+}
+
+impl Storage {
+    async fn load_schema_with_transaction(
+        &self,
+        tx: &mut Transaction<'_, Postgres>,
+    ) -> Result<IndexedPropertiesSchema, Error> {
+        let schema = sqlx::query_as::<_, (DocumentPropertyId, IndexedPropertyType)>(
+            "SELECT name, type FROM indexed_property_definition;",
+        )
+        .fetch_all(tx)
+        .await?
+        .into_iter()
+        .map(|(id, r#type)| (id, IndexedPropertyDefinition { r#type }))
+        .collect::<HashMap<_, _>>();
+
+        Ok(schema.into())
+    }
+
+    async fn extend_postgres_schema(
+        &self,
+        tx: &mut Transaction<'_, Postgres>,
+        update: &IndexedPropertiesSchemaUpdate,
+    ) -> Result<(), Error> {
```

Comment:
:art: similarly to the previous comment
```suggestion
    async fn extend_postgres_schema(
        tx: &mut Transaction<'_, Postgres>,
        update: &IndexedPropertiesSchemaUpdate,
    ) -> Result<(), Error> {
```